application/config/database.php

```diff
@@ -60,8 +60,15 @@
 $db['default']['db_debug'] = TRUE;
 $db['default']['cache_on'] = FALSE;
 $db['default']['cachedir'] = "";
-$db['default']['char_set'] = "utf8";
-$db['default']['dbcollat'] = "utf8_general_ci";
+
+if ($db['default']['dbdriver'] == "mysqli") {
+	$db['default']['char_set'] = "utf8mb4";
+	$db['default']['dbcollat'] = "utf8mb4_general_ci";
+}
+else {
+	$db['default']['char_set'] = "utf8";
+  $db['default']['dbcollat'] = "utf8_general_ci";
```

Comment:
As per ci3 doc, dbcollat is Only used in the ‘mysql’ and ‘mysqli’ drivers
https://codeigniter.com/userguide3/database/configuration.html?highlight=database#explanation-of-values

Reply:
dbcollat has been set by @back2arie for ~10 years for all db types (see [blame for current database.php](https://github.com/back2arie/Kalkun/blame/00e9ec16e27f8dfff553e768b9a0e593771114ed/application/config/database.php#L64)

@tenzap I've updated the code so that it's a little more obvious that dbcollat should only be used with the MySQL(i) driver.